app/src/main/java/crux/bphc/cms/fragments/MyCoursesFragment.java

```diff
@@ -462,6 +470,10 @@ public void setDownloadClickListener(ClickListener downloadClickListener) {
                             case 1:
                                 markAllAsRead(getLayoutPosition());
                                 break;
+
+                            case 2:
+                                markFavoriteStatus(getLayoutPosition(), isFavorite);
```

Comment:
I think you had the `!` on line 455 so as to invert the favorite status here? That defeats the purpose of naming the variable `isFavorite`. Invert it in the method call instead.